manifests/storage_state_crd.yaml

```diff
@@ -10,70 +12,69 @@ spec:
-          properties:
-            resource:
-              description: The resource this storageState is about.
-              type: object
-              properties:
-                group:
-                  description: The name of the group.
-                  type: string
-                resource:
-                  description: The name of the resource.
-                  type: string
-        status:
-          description: Status of the storage state.
-          type: object
-          properties:
-            currentStorageVersionHash:
-              description: The hash value of the current storage version, as shown
-                in the discovery document served by the API server. Storage Version
-                is the version to which objects are converted to before persisted.
-              type: string
-            lastHeartbeatTime:
-              description: LastHeartbeatTime is the last time the storage migration
-                triggering controller checks the storage version hash of this resource
-                in the discovery document and updates this field.
-              type: string
-              format: date-time
-            persistedStorageVersionHashes:
-              description: The hash values of storage versions that persisted instances
-                of spec.resource might still be encoded in. "Unknown" is a valid value
-                in the list, and is the default value. It is not safe to upgrade or
-                downgrade to an apiserver binary that does not support all versions
-                listed in this field, or if "Unknown" is listed. Once the storage
-                version migration for this resource has completed, the value of this
-                field is refined to only contain the currentStorageVersionHash. Once
-                the apiserver has changed the storage version, the new storage version
-                is appended to the list.
-              type: array
-              items:
+    subresources:
+      status: {}
+    validation:
+      openAPIV3Schema:
+        description: The state of the storage of a specific resource.
+        type: object
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
```

Comment:
```
The CustomResourceDefinition "storagestates.migration.k8s.io" is invalid: spec.validation.openAPIV3Schema.properties[metadata]: Forbidden: must not specify anything other than name and generateName, but metadata is implicitly specified
```
looking

Reply:
this will do the trick:
```
          metadata:
            type: object
            properties:
              name:
                description: The name must be "<.spec.resource.resouce>.<.spec.resource.group>".
                type: string
```

Reply:
filed https://github.com/kubernetes/kubernetes/issues/100933 in the main repo